client/src/main/java/io/pravega/client/connection/impl/TcpClientConnection.java

```diff
@@ -115,7 +113,7 @@ public void run() {
                     WireCommand command = readCommand(in, buffer);
                     if (command instanceof WireCommands.DataAppended) {
                         WireCommands.DataAppended dataAppended = (WireCommands.DataAppended) command;
-                        batchSizeTracker.recordAck(dataAppended.getEventNumber());
+                        flowToBatchSizeTracker.getAppendBatchSizeTrackerByFlowId(dataAppended.getRequestId()).recordAck(dataAppended.getEventNumber());
```

Comment:
A response to a conditionalAppend would create a AppendBatchSizetracker instance...

Reply:
Thanks @shrids, that's correct, but then we need to use `Flow.toFlowID()` in `CommandEncoder` as well as I added in the last commit. Otherwise, the `CommandEncoder` and `TcpClientConnection` refer to different keys for the same `AppendBatchSizeTracker`.